### ISSUE TITLE
Invisibility problem with android 4.0.4 Tablet

### ISSUE THREAD

**Issue by ahmetkucuk** [blazsolar/HorizontalPicker#14]
HorizontalPicker is not visible in the screen however when i touched the screen where HorizontalPicker suppose to be onItemClicked and onItemSelected methods are called. When i run it on Android L (Nexus 4) it is visible.

**Comment by gregorpichler** [comment]
I'm having the same problem on Android Wear.

**Comment by guillaumyle** [comment]
Same thing on Galaxy S2 with 4.0.3. 
More accurately, the text is invisible. 
Using a color or a selector doesn't change anything.

**Comment by phileo** [comment]
Not sure if this is related to the issue or not, but when the view becomes invisible, I see this in the logCat:

    E/libEGL: call to OpenGL ES API with no current context (Logged once per thread)
    W/HardwareRenderer: EGL error

**Comment by blazsolar** [comment]
@phileo Thanks for additional informations, i'll try to look into it as soon as possible.

**Comment by phileo** [comment]
could this be related:  https://code.google.com/p/android/issues/detail?id=37956
If so, which one of the 3 workarounds described would be most appropriate ?

**Comment by phileo** [comment]
found the problem!
Or at least a problem which affected my project, not 100% if this is the same problem.

In my case, the listener that I registered with mOnItemSelected was doing many things, and it seems some of those things are interfering with the OpenGL thread in canvas.draw().  So, instead of calling mOnItemSelected.onItemSelected() directly, I posted it to the UI Thread.  This appears to fix the invisibility issue in my case.

If you want, I can submit a pull request.
